### DISCUSSION TITLE
DIM AT with initialization

### DISCUSSION THREAD

**Discussion by matthewtillett** [gusmanb/ZXBasicStudio#56] [Q&A] [answered]
I can do this:
`DIM myArray(1,1) AS UBYTE AT 40000`

But cannot do this as the compiler complains with an error: Unexpected token '=>' <RIGHTARROW>
`DIM myArray(1,1) AS UBYTE AT 40000 => { {0,1}, {2,3} }`

Is there a way around this, but keeping the initialization?

**Comment by gusmanb** [comment, marked as answer]
Hi.

If it has not changed the compiler does not support to initialize arrays that are in a fixed address.

If this is for a big array I think what you could do is to create the array in the desired address and then LOAD a binary archive at that address, then it will be initialized with the content of the archive.

Cheers.

**Reply by matthewtillett** [reply]
Ahh, that sounds like a great idea!  I'll do some testing -- thanks for your help.
